File: Backend/src/main/java/com/corp/bookiki/bookinformation/entity/Category.java
package com.corp.bookiki.bookinformation.entity;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum Category {
    COMPUTER_SCIENCE(0, "컴퓨터과학"),
    PHILOSOPHY_PSYCHOLOGY(1, "철학/심리학"),
    RELIGION(2, "종교"),
    SOCIAL_SCIENCE(3, "사회과학"),
    LANGUAGE(4, "언어"),
    SCIENCE(5, "과학"),
    TECHNOLOGY(6, "기술"),
    ART(7, "예술"),
    LITERATURE(8, "문학"),
    HISTORY_GEOGRAPHY(9, "역사/지리");

    private final Integer code;
    private final String categoryName;

    // 숫자 -> enum
    public static Category ofCode(Integer code) {
        if (code == null) {
            return null;
        }

        for (Category category : Category.values()) {
            if (category.getCode().equals(code)) {
                return category;
            }
        }
        throw new IllegalArgumentException("Invalid category code: " + code);
    }

    // enum -> 숫자
    public static Integer toCode(Category category) {
        if (category == null) {
            return null;
        }
        return category.getCode();
    }
}

File: Frontend/src/types/enums/category.ts
export enum Category {
    COMPUTER_SCIENCE = 0,
    PHILOSOPHY_PSYCHOLOGY = 1,
    RELIGION = 2,
    SOCIAL_SCIENCE = 3,
    LANGUAGE = 4,
    SCIENCE = 5,
    TECHNOLOGY = 6,
    ART = 7,
    LITERATURE = 8,
    HISTORY_GEOGRAPHY = 9
  }
  
  export const CategoryNames: Record<Category, string> = {
    [Category.COMPUTER_SCIENCE]: '컴퓨터과학',
    [Category.PHILOSOPHY_PSYCHOLOGY]: '철학/심리학',
    [Category.RELIGION]: '종교',
    [Category.SOCIAL_SCIENCE]: '사회과학',
    [Category.LANGUAGE]: '언어',
    [Category.SCIENCE]: '과학',
    [Category.TECHNOLOGY]: '기술',
    [Category.ART]: '예술',
    [Category.LITERATURE]: '문학',
    [Category.HISTORY_GEOGRAPHY]: '역사/지리'
  }
  
  // 유틸리티 함수들
  export const CategoryUtils = {
    // number -> Category
    ofCode(code: number | null): Category | null {
      if (code === null) return null
      if (Object.values(Category).includes(code)) {
        return code as Category
      }
      throw new Error(`Invalid category code: ${code}`)
    },
  
    // Category -> string
    getCategoryName(category: Category | null): string | null {
      if (category === null) return null
      return CategoryNames[category]
    }
  }